### DISCUSSION TITLE
Handle large Asset uploads

### DISCUSSION THREAD

**Discussion by angrybrad** [craftcms/cms#9084] [Ideas]
> Created by: Fred Carlsen (fred@sjelfull.no) on 2015/03/11 07:34:22 +0000
> Votes at time of UserVoice import: 3

-----------------------------------

It would be great if Asset supported chunked and resumable uploads.

Some clients need to upload large files, and it's generally not a good idea to up those max_upload_size limits.

**Comment by maknapp** [comment]
I would suggest looking at https://tus.io/ for implementing file uploading. This would remove all file size upload limits while also reducing memory usage for all file uploads in general. Streaming uploads would allow for faster uploading to 3rd party volumes like AWS - craft could pass the file along as it receives it instead of waiting for the entire file before sending to the volume. File uploading via form POST should be removed entirely as there is no upside.

**Comment by piotrpog** [comment]
Implementing chunked uploads would be really useful. Please consider it for Craft 4 or even 3.5 :)

**Comment by andris-sevcenko** [comment]
This is definitely worth considering, however, [tus-php](https://github.com/ankitpokhrel/tus-php) currently requires PHP 7.1.3, so the first step is getting to a point where Craft requires something more recent than PHP 7.0.

**Comment by lassemt** [comment]
Would tus-php be an better feature for the S3-plugin and not in the core? As one probably don't want to upload big files to a server? Might be an releavnt update now since Craft’s minimum PHP version requirement was increased to 7.2.5 in 3.6.

**Comment by andris-sevcenko** [comment]
Worth re-visiting for 4.0 and considering it. Unlikely there will be Craft 3.7 and we're mixing things up with Volumes under the hood in 4.0.

**Comment by maknapp** [comment]
@brandonkelly Is large file uploading completed for v4 or closed as a "wontfix" issue?

**Reply by maknapp** [reply]
Looking at [Uploader.js](https://github.com/craftcms/cms/blob/4.0.0.1/src/web/assets/cp/src/js/Uploader.js), it is still uploading via form POST, so I am going to guess this not completed.

**Reply by brandonkelly** [reply]
We unfortunately didn’t get around to this yet, but it’s on the list.

**Comment by myriago** [comment]
When using Cloudflare's free plan, file upload size is limited to 100MB. Paid plans offer slightly larger limits. Chunk file upload would help here. While there's an abandoned plugin for v4, it's limited to the Assets page and doesn't work on the Entries page.
